project kickoff

josepplloo opened the pull request:
# Summary
  - New component Title added.
  - New component Input added.
  - New Unordered list for the group of items added. 
 
 > TODO: Finish the styles and the test in the unordered list.
  ## Screenshot
![image](https://user-images.githubusercontent.com/1114740/74500369-ab1b6300-4eb4-11ea-96d3-8352f1ec5131.png)

Commit:
Initial testing configuration

Commit:
adding input element and some global css styles

Commit:
adding input element and some global css styles

josepplloo committed:
adding new styles to the input component

josepplloo committed:
fixing styles in input element and adding the UL element

josepplloo committed:
fixing styles in input element and addingadding the file elements

Commit:
Testing the new item Component

Commit:
Testing the new item Component

josepplloo committed:
Fixing issues related with the render of LI items in createTodoItem

anyruizd commented on src/components/App/app.scss:
```diff
@@ -0,0 +1,19 @@
+@import url('https://fonts.googleapis.com/css?family=Raleway:100,100i,200,200i,300,300i,400,400i,500,500i,600,600i,700,700i,800,800i,900,900i&display=swap');
```
Are you sure you need to import all those font sizes? This adds a lot of weight to your app.

anyruizd replied:
Besides, is there a special reason to import fonts here and not use a ‘link‘ element in your html header?

josepplloo replied:
Nope, `<link>` is enough

anyruizd commented on src/components/Input/constants.js:
```diff
@@ -0,0 +1,4 @@
+export default {
+  nameplaceText: ' Write something you need to do',
```
Wouldn’t it be better something like placeholderText instead of nameplaceText?

anyruizd commented on src/components/App/createApp.js:
```diff
@@ -0,0 +1,16 @@
+import createItemList from '../ItemList/createItemList';
+import './app.scss';
+
+
```
Please erase this additional blank line.

anyruizd commented on src/components/App/app.scss:
```diff
@@ -0,0 +1,19 @@
+
+body {
+  font-size: 16px;
+  font-family: 'Raleway', sans-serif;
```
Wouldn’t it be better to declare the font properties as variables?

josepplloo replied:
Yes, it will be done in the future PR 👐

anyruizd commented on src/components/Input/createInput.test.js:
```diff
@@ -0,0 +1,13 @@
+    expect(inputElement.tagName).toBe('INPUT');
+  });
+
+  it('should be like the snaptshot', () => {
```
It would be nice to have more complete description, in this way, if I read the test description I won't need to go to the snapshot to see what was tested. Try to answer the questions: What specifically am I testing here? What properties?

josepplloo replied:
Yes, It consider layout and class name as the properties to test.

anyruizd commented on src/components/ItemList/createItemList.js:
```diff
@@ -0,0 +1,14 @@
+import constants from './constants';
+import './itemList.scss';
```
Try to group the imports by category. In this case for instance, the `.scss` import should go at the end.

anyruizd commented on src/components/ItemList/createItemList.test.js:
```diff
@@ -0,0 +1,12 @@
+import createItemList from './createItemList';
+
+describe('test for ItemList component', () => {
```
Since you are testing each component in its own separate file, I don't think it is necessary to add a `describe()` in none case.

anyruizd commented on src/components/TodoItem/createTodoItem.js:
```diff
@@ -0,0 +1,8 @@
+function createTodoItem({ itemName, itemStatus } = { itemName: '', itemStatus: false }, index = 0) {
```
For readability, your code lines shouldn't have more than 80 characters long, if this is the case, consider breaking lines.

josepplloo committed:
Summary
  - describe test suite wrapper test function erased.
  - using link tag instead @import to load the fonts.
  - erasing unnecessary blank spaces

josepplloo committed:
change variable name for placeholder input text

josepplloo committed:
erasing unused fonts
